### PR TITLE
Add rotating token support for admin API

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Add rotating token support for admin API

## Motivation and Context
Use the *credentials.Credentials implementation method *Get*

```
func (c *Credentials) Get() (Value, error) {
```

which also handles auto-refresh, this allows for chaining
of various implementations together if necessary or simply
initialize with credentials.NewStaticV4(access, secret, token)

## How to test this PR?
You can initialize the admin API with madmin.NewOptions()

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Review by Alevsk — APPROVED** [review]

File: pkg/madmin/api.go [hunk 1]
```diff
@@ -85,38 +87,67 @@ const (
...
+	// Save endpoint URL, user agent for future uses.
+	clnt.endpointURL = endpointURL
+
+	// Instantiate http client and bucket location cache.
+	clnt.httpClient = &http.Client{
+		Jar:       jar,
+		Transport: DefaultTransport(secure),
 	}
 
+	// Introduce a new locked random seed.
```
Comment (klauspost) [inline pkg/madmin/api.go]:
```suggestion
	// Add locked pseudo-random number generator.
```

**Review by klauspost — APPROVED** [review]
Just a doc update, so people don't expect it to be crypto-safe.  (I see it is only used for retry jitter)

File: pkg/madmin/api.go [hunk 2]
```diff
@@ -85,38 +87,67 @@ const (
...
+	if err != nil {
+		return nil, err
+	}
+	return clnt, nil
+}
+
+func privateNew(endpoint string, creds *credentials.Credentials, secure bool) (*AdminClient, error) {
+	// Initialize cookies to preserve server sent cookies if any and replay
+	// them upon each request.
+	jar, err := cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
```
Comment (vadmeste) [inline pkg/madmin/api.go:120]: This is an admin client, not sure why cookiejar is being used here, could you add a comment to explain ?

**Review by vadmeste — COMMENTED** [review]
Only one comment, LGTM otherwise

File: pkg/madmin/api.go (hunk 2 above)
Comment (harshavardhana) [reply pkg/madmin/api.go:120]: The comment is above @vadmeste

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#9fed6cfb-4b37-4016-b22f-b8a7a7108767) |
<a name="9fed6cfb-4b37-4016-b22f-b8a7a7108767"></a>
### 9244-06c5387/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32495
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b9ba61bff70c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 8 minutes and 50 seconds
(4/15) Running aws-sdk-ruby tests ... done in 20 seconds
(5/15) Running awscli tests ... done in 2 minutes and 57 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 6 minutes and 17 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 44 seconds
(9/15) Running minio-go tests ... done in 19 minutes and 1 seconds
(10/15) Running minio-java tests ... FAILED in 8 minutes and 28 seconds
{
  "name": "minio-java",
  "function": "composeObject(String bucketName, String objectName,List<ComposeSource> composeSources, Map <String,String > headerMap, ServerSideEncryption sseTarget)",
  "args": "size: 6 MB & 6 MB",
  "duration": 11516,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=minio-java-test-21hgcj2, objectName=minio-java-test-1io0s50, resource=/minio-java-test-21hgcj2/minio-java-test-1io0s50, requestId=1601CC3CD9A83ECE, hostId=a7d5181c-a3c4-412d-91cc-0bc183935fc9)\nrequest={method=PUT, url=http://minio-dev7.minio.io:32495/minio-java-test-21hgcj2/minio-java-test-1io0s50?uploadId=e669fd3f7e03e28e&partNumber=1, headers=x-amz-copy-source-if-match: 7aca5dd3dbad98c0d7180e14426210ce-2\nx-amz-copy-source: minio-java-test-21hgcj2/minio-java-test-2j9sv75\nAccept-Encoding: identity\nHost: minio-dev7.minio.io:32495\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200401T203145Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200401/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=501, headers=Accept-Ranges: bytes\nContent-Length: 410\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 1601CC3CD9A83ECE\nX-Xss-Protection: 1; mode=block\nDate: Wed, 01 Apr 2020 20:31:45 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1220), io.minio.MinioClient.execute(MinioClient.java:1082), io.minio.MinioClient.executePut(MinioClient.java:1446), io.minio.MinioClient.executePut(MinioClient.java:1468), io.minio.MinioClient.uploadPartCopy(MinioClient.java:2619), io.minio.MinioClient.composeObject(MinioClient.java:2572), FunctionalTest.composeObject_test1(FunctionalTest.java:2587), FunctionalTest.runTests(FunctionalTest.java:3396), FunctionalTest.main(FunctionalTest.java:3514)]"
}

Executed 9 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**